Sources/SwiftSyntaxBuilderGeneration/Templates/TokensFile.swift

```diff
@@ -18,22 +18,14 @@ let tokensFile = SourceFile {
   ImportDecl(importTok: TokenSyntax.import.withLeadingTrivia(.docLineComment(copyrightHeader)), path: "SwiftSyntax")
 
   ExtensionDecl(
+    modifiers: ModifierList([TokenSyntax.public.withLeadingTrivia(.newlines(1) + .docLineComment("/// Namespace for commonly used tokens with default trivia.") + .newlines(1))]),
```

Comment:
Also, we might want to extract these `.withLeadingTrivia`s, rather than 'attaching' them to the first modifier token. I haven't dug too deep into this yet, perhaps we could extend constructs like `ExtensionDecl` to have a `withLeadingTrivia` method?

```swift
SourceFile {
  ...
  ExtensionDecl(
    ...
  )
  .withLeadingTrivia(.newlines(1) + .docLineComment(...) + .newlines(1))
}
```

Or perhaps declare them separately, something very roughly along the lines of

```swift
SourceFile {
  ...
  Trivia([.newlines(1) + .docLineComment(...) + .newlines(1)])
  ExtensionDecl(
    ...
  )
}
```

(straw-man syntax, I haven't checked whether we could actually do something like this)

Anyway, this is probably something for another PR.

Reply:
I think with the current design your first proposal is easier to implement for now, the second one looks promising but there are a few challenges like deciding whether to attach the trivia as leading trivia to `ExtensionDecl` or as trailing trivia to the previous node.

@iMostfa started working on your first proposal in https://github.com/apple/swift-syntax/pull/385. Maybe you could pick up from there, but I agree that this is something for a follow-up PR.

Reply:
Interesting, I didn't know about the other PR. For the second idea, perhaps something like

```swift
SourceFile {
  ...
  LeadingTrivia(...)
  ExtensionDecl(...)
  TrailingTrivia(...)
  ...
}
```

would be possible, having the builder attach the trivia to the adjacent node internally. This way the builder closure would closely resemble the generated code.

Reply:
That sounds like an interesting idea. We need to figure out how to express this in the type system, but that should be possible with a couple more protocols.